# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/tests/gtest/TestMultiplexInputStream.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d xpcom/tests/gtest/TestMultiplexInputStream.cpp
--- a/xpcom/tests/gtest/TestMultiplexInputStream.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/xpcom/tests/gtest/TestMultiplexInputStream.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -14,7 +14,7 @@
 #include "nsThreadUtils.h"
 #include "Helpers.h"
 
-using mozilla::GetCurrentThreadSerialEventTarget;
+using mozilla::GetCurrentSerialEventTarget;
 using mozilla::SpinEventLoopUntil;
 
 TEST(MultiplexInputStream, Seek_SET)
@@ -372,7 +372,7 @@
   ASSERT_TRUE(cb->Called());
 }
 
-// AsyncWait - withEventTarget - closureOnly
+// AsyncWait - with EventTarget - closureOnly
 TEST(MultiplexInputStream, AsyncWait_withEventTarget_closureOnly)
 {
   nsCOMPtr<nsIInputStream> is = CreateStreamHelper();
@@ -771,7 +771,7 @@
 
   // Let's read the size async.
   RefPtr<testing::LengthCallback> callback = new testing::LengthCallback();
-  rv = afsis->AsyncLengthWait(callback, GetCurrentThreadSerialEventTarget());
+  rv = afsis->AsyncLengthWait(callback, GetCurrentSerialEventTarget());
   ASSERT_EQ(NS_OK, rv);
 
   MOZ_ALWAYS_TRUE(SpinEventLoopUntil([&]() { return callback->Called(); }));
@@ -799,11 +799,11 @@
 
   // Let's read the size async.
   RefPtr<testing::LengthCallback> callback1 = new testing::LengthCallback();
-  rv = afsis->AsyncLengthWait(callback1, GetCurrentThreadSerialEventTarget());
+  rv = afsis->AsyncLengthWait(callback1, GetCurrentSerialEventTarget());
   ASSERT_EQ(NS_OK, rv);
 
   RefPtr<testing::LengthCallback> callback2 = new testing::LengthCallback();
-  rv = afsis->AsyncLengthWait(callback2, GetCurrentThreadSerialEventTarget());
+  rv = afsis->AsyncLengthWait(callback2, GetCurrentSerialEventTarget());
   ASSERT_EQ(NS_OK, rv);
 
   MOZ_ALWAYS_TRUE(SpinEventLoopUntil([&]() { return callback2->Called(); }));